pkg/agent/controller/egress/egress_controller.go

```diff
@@ -613,29 +614,78 @@ func (c *EgressController) unbindPodEgress(pod, egress string) (string, bool) {
 	return "", false
 }
 
-func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP string) error {
+func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP string, scheduleErr error) error {
 	isLocal := false
 	if egressIP != "" {
 		isLocal = c.localIPDetector.IsLocalIP(egressIP)
 	}
+
+	desiredStatus := &crdv1b1.EgressStatus{}
+	if isLocal {
+		desiredStatus.EgressNode = c.nodeName
+		desiredStatus.EgressIP = egressIP
+		if isEgressSchedulable(egress) {
+			desiredStatus.Conditions = []crdv1b1.EgressCondition{
+				{
+					Type:               crdv1b1.IPAssigned,
+					Status:             corev1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
+					Reason:             "Assigned",
+					Message:            "EgressIP is successfully assigned to EgressNode",
+				},
+			}
+		}
+	} else if egressIP == "" {
+		// Select one Node to update false status among all Nodes.
+		// We don't care the value of egress.Spec.EgressIP, just use it to reach a consensus among all agents about
```

Comment:
> We don't care about

Reply:
Fixed, hope it’s the last time I made the same mistake :)